avae/train.py

```diff
@@ -287,6 +295,10 @@ def train(
                 xs, ys, vae, device, poses=ps  # do we need val and test here?
             )
 
+    if config.VIS_HIS:
```

Comment:
Apologies if I had missed something, but this appears to execute at the end of the run (i.e. after all epochs are completed). If training fails or is stopped, this file will never be created. Since this is only required once per run, I suggest you move this earlier in the code to ensure it gets created. 

In fact, since this is to do with data statistics, maybe a better place for this call would be in `load_data` function (e.g. line 40)  when other data statistics are printed and other one-time figures plotted (e.g. affinity matrix). `vis` module is already imported there so it doesn't add any new imports.

Reply:
good point, implemented in [1faf704](https://github.com/alan-turing-institute/affinity-vae/pull/69/commits/1faf704d43a288078d8aaedc1715f1861fe729b9). After test, val split.